sdk/communication/AzureCommunicationChat/Source/Signaling/CommunicationSignalingClient.swift

```diff
@@ -149,8 +149,40 @@ class CommunicationHandler: NSObject, TrouterListener {
             let chatEventId = try ChatEventId(forCode: generalPayload._eventId)
 
             // Convert trouter payload to chat event payload
-            let chatEvent = try TrouterEventUtil.create(chatEvent: chatEventId, from: request)
-            handler(chatEvent, chatEventId)
+            let eventType = TrouterEventUtil.create(chatEvent: chatEventId, from: request)
+
+            switch eventType {
+            case EventType.chatMessageReceivedEvent:
+                let event = try ChatMessageReceivedEvent(from: request)
+                handler(event, chatEventId)
```

Comment:
Why do you not simply switch on chatEventId?  The conversion to EventType adds no value.